### DISCUSSION TITLE
Client Device Heterogeneity

### DISCUSSION THREAD

**Discussion by Pedja-Djape** [NVIDIA/NVFlare#1054] [Q&A] [answered]
Hi all,

1.      I have looked into the FLARE documentation and scanned through the discussions on this repository, but I can't seem to come to a conclusive answer to whether or not FLARE is client device-agnostic? More specifically, I have a research project and would like to conduct FL via mobile devices: Does FLARE support deployment on mobile phones?
2. Can anyone share some demos or tutorials on how to implement vertical federated learning?

Thank you for the help.

**Comment by holgerroth** [comment, marked as answer]
@chesterxgchen regarding 1.

Regarding 2., you can implement vertical FL workflow using the controller API. See existing examples [here](https://github.com/NVIDIA/NVFlare/tree/dev/nvflare/app_common/workflows).
We will provide VFL examples/tutorials in the near future. See also https://github.com/NVIDIA/NVFlare/discussions/793.

**Reply by Pedja-Djape** [reply]
Hi @chesterxgchen do you know if FLARE is client device-agnostic or usable with mobiles?

**Reply by chesterxgchen** [reply]
Sorry for the late response.  NVFLARE has no assumption about client device. That being said, we haven't worked with mobile devices. 
Although it is possible, we haven't tried. There are probably other factors to consider beside type of the client's devices, such as communication protocol, communication/message patterns.

**Comment by YuanTingHsieh** [comment]
@Pedja-Djape vertical learning example is available at: https://github.com/NVIDIA/NVFlare/tree/dev/examples/advanced/vertical_federated_learning
